lnrpc/routerrpc/router_server.go

```diff
@@ -425,7 +425,16 @@ func marshallError(sendError error) (*Failure, error) {
 		return nil, fmt.Errorf("cannot marshall failure %T", onionErr)
 	}
 
-	response.FailureSourceIndex = uint32(fErr.FailureSourceIdx)
+	// If the ClearTextError received is a ForwardingError, the error
+	// originated from a node along the route, not locally on our outgoing
+	// link. We set failureSourceIdx to the index of the node where the
+	// failure occurred. If the error is not a ForwardingError, the failure
+	// occurred at our node, so we leave the index as 0 to indicate that
+	// we failed locally.
+	fErr, ok := rtErr.(*htlcswitch.ForwardingError)
```

Comment:
Do we always assume 0 for the other error types? If so the index could be part of the ClearTextError iface to avoid the type conversions. I see though we only do so in two places, so may not be worth the effort now

Reply:
Yes, only other type is a `LinkError` which implies a 0 index. Going to leave as is for now, but will keep it in mind!